client/src/main/java/org/eclipse/hono/client/impl/AbstractRequestResponseClient.java

```diff
@@ -459,7 +460,12 @@ protected final void handleResponse(final ProtonDelivery delivery, final Message
             if (response == null) {
                 LOG.debug("discarding malformed response lacking status code [reply-to: {}, correlation ID: {}]",
                         replyToAddress, message.getCorrelationId());
-                TracingHelper.logError(span, "response from peer released (no status code)");
+                if (span != null) {
+                    TracingHelper.logError(span, "response from peer released (no status code)");
+                    Tags.HTTP_STATUS.set(span, HttpURLConnection.HTTP_INTERNAL_ERROR);
```

Comment:
hmm, receiving a `501 Not Implemented` response could distract from a possible scenario where a bug in the component on the receiving end has caused the missing status code. Therefore I'd rather use the more generic "Internal Server Error".

Reply:
yes. I meant the peer that the `AbstractRequestResponseClient` is sending requests to.

Reply:
I would argue that this will be the sole source of the problem in all cases, not just a *possible scenario*. This is the reason why I proposed to use 501, because it is the remote service's fault that the invocation failed. In most cases 500 is used to indicate an unexpected problem of unknown origin (e.g. an uncaught runtime exception) which is not the case here. WDYT?

Reply:
I see what you mean that the "Internal Server Error" exception might not sufficiently point to the remote service as the source of the exception.
But I think an "Not Implemented" error wouldn't really point me there either.
And I would associate "Not Implemented" rather with "No bug here per se - feature just not implemented", so I would think I have sent an unsupported request with the `AbstractRequestResponseClient`, rather than thinking this is a bug in a dependent service that should be investigated/reported.
How about using 500 with a better message:
````
handler.one().handle(Future.failedFuture(new ServerErrorException(HttpURLConnection.HTTP_INTERNAL_ERROR,
        "malformed response from " + getName() + " endpoint (no status code)")));
````

Reply:
I see your point and I think your proposal is good :-) However, I think we should omit the `(no status code)` because that is, in fact, only one possible problem with the response.

Reply:
ok, I've removed the `(no status code)` and added an extra log line where the status is checked.